waspc/src/Wasp/Analyzer/TypeChecker/AST.hs

```diff
@@ -4,30 +4,38 @@ module Wasp.Analyzer.TypeChecker.AST
     TypedExpr (..),
     Identifier,
     ExtImportName (..),
+    WithCtx (..),
+    withCtx,
     exprType,
   )
 where
 
 import Wasp.Analyzer.Parser (ExtImportName (..), Identifier)
+import Wasp.Analyzer.Parser.Ctx (WithCtx (..), withCtx)
 import Wasp.Analyzer.Type
 
-newtype TypedAST = TypedAST {typedStmts :: [TypedStmt]} deriving (Eq, Show)
+newtype TypedAST = TypedAST {typedStmts :: [WithCtx TypedStmt]} deriving (Eq, Show)
```

Comment:
I used the same trick here as in Parser.AST -> used WithCtx to embed Ctx in correct places.